### ISSUE TITLE
provide signature help for local functions inside static method

### ISSUE THREAD

**Issue by ivanbasov** [dotnet/roslyn#38181]
The following scenario is not working:

```
   public async Task TestLocalFunctionInStaticMethod() 
         { 
             var markup = @" 
 class C 
 { 
     static void M() 
     { 
         void Local() { } 
  
         Local($$); 
     } 
 }"; 
  
             var expectedOrderedItems = new List<SignatureHelpTestItem> { new SignatureHelpTestItem("void Local()") }; 
  
             await TestAsync(markup, expectedOrderedItems); 
         } 
```

Here is the discussion: https://github.com/dotnet/roslyn/issues/38074#issuecomment-522800766

It seems that the issue is in the semantic model. We need to use 
https://github.com/dotnet/roslyn/blob/5551fad9651b91dcc64a965cfd93e1da4f386273/src/Features/CSharp/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider_MethodGroup.cs#L56-L60

`RequiresInstanceReceiver` instead of `!IsStatic` but it is unavailable. See https://github.com/dotnet/roslyn/pull/38117#issuecomment-523613605

**Comment by ivanbasov** [comment]
resolved in 16.4.P2 with https://github.com/dotnet/roslyn/pull/38117

**Comment by ivanbasov** [comment]
There is a signature help part at https://github.com/dotnet/roslyn/pull/38811 to be merged.
